### PR TITLE
Migrate to Spring 6

### PR DESCRIPTION
**Pull request by vkalapov**

### CODE REVIEW

**Review by ikasarov — APPROVED** [review]

File: pom.xml [hunk 1]
```diff
@@ -18,7 +18,7 @@
         <mockito.version>5.11.0</mockito.version>
         <jackson.version>2.17.0</jackson.version>
         <jackson.databind.version>2.17.0</jackson.databind.version>
-        <spring.version>5.3.34</spring.version>
+        <spring.version>6.1.10</spring.version>
```
Comment (IvanBorislavovDimitrov) [inline pom.xml:21]: there is a newer version, let's use it

File: src/main/java/com/sap/cloudfoundry/client/facade/rest/CloudControllerResponseErrorHandler.java [hunk 2]
```diff
@@ -63,7 +62,8 @@ private static String concatenateErrorMessages(List<Map<String, Object>> errors)
 
     @Override
     public void handleError(ClientHttpResponse response) throws IOException {
-        HttpStatus statusCode = response.getStatusCode();
+        HttpStatus statusCode = HttpStatus.valueOf(response.getStatusCode()
+                                                           .value());
```
Comment (IvanBorislavovDimitrov) [inline src/main/java/com/sap/cloudfoundry/client/facade/rest/CloudControllerResponseErrorHandler.java:66]: why is it needed to use valueOf

Comment (vkalapov) [reply src/main/java/com/sap/cloudfoundry/client/facade/rest/CloudControllerResponseErrorHandler.java:66]: response.getStatusCode() returns HttpStatusCode in Spring 6, hence why the conversion is needed

File: pom.xml (hunk 1 above)
Comment (vkalapov) [reply pom.xml]: done

**Review by IvanBorislavovDimitrov — APPROVED** [review]

**Pull request merged**